BUILD.md

```diff
@@ -557,3 +557,19 @@ Build and generate report:
 meson setup build
 ninja -C build scan-build
 ```
+
+### Installation
+
+You can specify directory where files will be installed using `DESTDIR`
+environment variable, and configure prefix and directory paths using `setup`
+parameters:
+
+```shell
+meson setup --prefix=/usr build
+DESTDIR="$PWD/dest" ninja -C build install
+# files will be installed in: dest/usr/bin, dest/usr/share, etc…
+```
```

Comment:
The [meson_doc] say that special things happen when prefix is **/usr** as well as **/usr/local**, so the helpful example pattern `files will be installed in: dest/usr/bin, dest/usr/share, etc…` maybe not be correct.

For example, when the prefix is `/usr`:
- sysconfdir defaults to `/etc`
- localstatedir defaults to `/var`
- sharedstatedir defaults to `/var/lib`

When DESTDIR is added to the mix, the install directories become `dest/usr`, `dest/etc`, `dest/var`, and `dest/var/lib`.  Same situation for `/usr/local`:

When the prefix is` /usr/local`:
 - localstatedir defaults to `/var/local`
 - sharedstatedir defaults to `/var/local/lib`

When DESTDIR is added to the mix, the install directories become `dest/usr/local`, `dest/var/local`, and `dest/var/local/lib`. 

That's assuming DESTDIR is a wrapper around the prefix logic, and that the above prefix "top-level-overrides" aren't also taking DESTDIR into account. 

I think anyone that's run `./confgure --prefix=` understands the pattern and know what they'll get after a `make install`.

But I have to admit that mingling `$DESTDIR` **_and_**  `$prefix` combined with the above nuances turns what was a _no-brainer_ into something _I-really-need-to-think-about_ to avoid polluting my system or doing something I didn't expect.  

Ideally meson would have a one or two lines describing the "99%-of-cases" to help users install either to their home `~/.local/..` directory or `/usr/local`, which I believe is the 99% scenario.  After that, we'd point package maintainers to the deep-docs.  But I'm having trouble finding that short snippet?

Reply:
`DESTDIR` is not my invention - I merely documented it. It's a separate thing from prefix:

- prefix, sysconfdir, etc is for paths intended on end-users' machines
- DESTDIR is intended for build directory on packager's machine

Here's DESTDIR [documentation for automake](https://www.gnu.org/software/automake/manual/html_node/DESTDIR.html) (but I think this variable predates automake). 

Maybe I should remove this from our documentation altogether? I mean: it works in Meson exactly the same way it worked in Automake, or CMake, or other buildsystems.

I don't want to re-write already available Meson documentation for things that are obvious… So perhaps I should get rid of this section altogether and create a new task for meson migration: writing file PACKAGING.md?

Reply:
> DESTDIR is not my invention 

Yup, totally understand that you're passing along "DESTDIR".  

> I should get rid of this section altogether and create a new task for meson migration: writing file PACKAGING.md?

See my examples for Windows/Mac/Linux in comment below -- I think we should keep something at that level (ie: easy-to-copy and paste recommended installation paths for general users) up in this file.

For those creating OS-grade packages or power-users bundling their own DEB/RPM/etc.. I think breaking out to a detailed `PACKAGING.md` is perfect.

I know personally, I'm not aware of all the nuances involved in generating OS-type package bundles, and I'd rather not be bogged down with those details in a basic user-facing document like this.